Giga Coaster design name display wrong (fixes #5077)

raphaelsetin opened the pull request:
In this pull request we are addressing the "Goliath.1" issue #5077 that is in OpenRCT2. This fix addresses better than the solution proposed in #5337, because ours is more specific. We only change the displayed name of the track "Goliath.1" to "Goliath" by an if-statement. Therefore our solution is more specific to the issue.

Gymnasiast commented:
I'm afraid this solution is too limited. What it should do is the following:

When loading a track file, set its name to everything before the first dot.

The above is how RCT2 behaved and it allows people to have multiple track designs that carry the same name. Goliath.TD6 and Goliath.1.TD6 are just one example.

raphaelsetin commented:
Hey Michael!

But if I take everything before the dot as a name, then there will be 2 Goliath track names, right? Even if not, don't you think that getting everything before the dot can be bad if for instance, you guys release a track name called "Mega.Rollercoaster.TD6"? That's why I didn't do in that way, however if you want me to do it, I can do it and update the pull request for you. Let me know!

Thank you.

--
[cid:storage_emulated_0_Download_Signature_png_1492284319914]


-------- Original message --------
From: Michael Steenbeek <notifications@github.com>
Date: 4/18/2017 4:36 AM (GMT-05:00)
To: OpenRCT2/OpenRCT2 <OpenRCT2@noreply.github.com>
Cc: Raphael Setin <raphael-setin@hotmail.com>, Author <author@noreply.github.com>
Subject: Re: [OpenRCT2/OpenRCT2] Giga Coaster design name display wrong (fixes #5077) (#5380)


I'm afraid this solution is too limited. What it should do is the following:

When loading a track file, set its name to everything before the first dot.

The above is how RCT2 behaved and it allows people to have multiple track designs that carry the same name. Goliath.TD6 and Goliath.1.TD6 are just one example.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/OpenRCT2/OpenRCT2/pull/5380#issuecomment-294730470>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AYgGJCwDPXFPrPZZpk1KzpxM1Al8xKSnks5rxHYogaJpZM4M_aAu>.

raphaelsetin commented:
Never mind @Gymnasiast , I already corrected the code in my end and sooner I will delete my old commit and add this new solution that I just did. My new solution corrects the problem in the rides/tracks function only, and sets the name of the track as everything in the filename that is right before the dot (which was what you proposed to me).

raphaelsetin commented:
@Gymnasiast I just added the new commit and rebased everything. Tell me if you think this is what you were looking for. I did exactly what you instructed, so I think I should be fine, but let me know if otherwise. Thank you.

rwjuk commented:
Well, I tested this and it worked as I expected it to.

Gymnasiast commented on src/openrct2/core/Path.cpp:
```diff
@@ -104,6 +104,7 @@ namespace Path
         utf8 * cstr = GetFileNameWithoutExtension(path.c_str());
         std::string result = String::ToStd(cstr);
         Memory::Free(cstr);
+        
```
Could you remove this line, so that this file does not show up as changed?

Gymnasiast commented on src/openrct2/ride/TrackDesignRepository.cpp:
```diff
@@ -346,7 +346,10 @@ class TrackDesignRepository final : public ITrackDesignRepository
     static std::string GetNameFromTrackPath(const std::string &path)
     {
-        return Path::GetFileNameWithoutExtension(path);
+		std::string name = Path::GetFileNameWithoutExtension(path);
```
Please replace the tabs with spaces. We use spaces in C++ files and tabs in C files (which is confusing, I admit, but we're working on that).

raphaelsetin commented:
@Gymnasiast I just made the corrections! Let me know if that's good for you. Thanks again!
